web/hooks/pull_request.go

```diff
@@ -0,0 +1,198 @@
+package hooks
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	pb "github.com/autograde/quickfeed/ag"
+	"github.com/google/go-github/v35/github"
+	"gorm.io/gorm"
+)
+
+func (wh GitHubWebHook) handlePullRequestReview(payload *github.PullRequestReviewEvent) {
+	wh.logger.Debugf("Received pull request review event for pull request: %s, in repository: %s",
+		payload.GetPullRequest().GetTitle(), payload.GetRepo().GetName())
+
+	// Currently, QF only needs to do something if the PR is approved
+	if payload.GetReview().GetState() != "approved" {
+		wh.logger.Debug("Ignoring pull request review event for non-approved review")
+		return
+	}
+	// We make sure that the pull request is one that QF has a data record of
+	pullRequest, err := wh.db.GetPullRequest(&pb.PullRequest{
+		ExternalRepositoryID: uint64(payload.GetRepo().GetID()),
+		Number:               uint64(payload.GetPullRequest().GetNumber()),
+	})
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			wh.logger.Debugf("Ignoring pull request review event for non-managed pull request #%d, in %s",
+				payload.GetPullRequest().GetNumber(), payload.GetRepo().GetFullName())
+		} else {
+			wh.logger.Errorf("Failed to get pull request from database %v", err)
```

Comment:
Maybe add PR number and repo as in the debug line above.

Reply:
On second thought, the debug line at the top of the method should contain enough context.